xbmc/windowing/gbm/WinSystemGbm.cpp

```diff
@@ -271,3 +272,153 @@ std::unique_ptr<CVideoSync> CWinSystemGbm::GetVideoSync(void* clock)
 {
   return std::make_unique<CVideoSyncGbm>(clock);
 }
+
+bool CWinSystemGbm::SetHDR(const VideoPicture* videoPicture)
+{
+  auto settingsComponent = CServiceBroker::GetSettingsComponent();
+  if (!settingsComponent)
+    return false;
+
+  auto settings = settingsComponent->GetSettings();
+  if (!settings)
+    return false;
+
+  if (!settings->GetBool(SETTING_WINSYSTEM_IS_HDR_DISPLAY))
+    return false;
+
+  auto drm = std::dynamic_pointer_cast<CDRMAtomic>(m_DRM);
+  if (!drm)
+    return false;
+
+  if (!videoPicture)
+  {
+    auto connector = drm->GetConnector();
+    if (connector->SupportsProperty("HDR_OUTPUT_METADATA"))
+    {
+      drm->AddProperty(connector, "HDR_OUTPUT_METADATA", 0);
+      drm->SetActive(true);
+
+      if (m_hdr_blob_id)
+        drmModeDestroyPropertyBlob(drm->GetFileDescriptor(), m_hdr_blob_id);
+      m_hdr_blob_id = 0;
+    }
+
+    return true;
+  }
+
+  // Try video plane first and fallback to gui plane
+  auto plane = drm->GetVideoPlane();
+
+  if (!plane)
+    plane = drm->GetGuiPlane();
+
+  if (!plane)
+    return false;
+
+  bool result;
+  uint64_t value;
+  std::tie(result, value) =
+      plane->GetPropertyValue("COLOR_ENCODING", DRMPRIME::GetColorEncoding(*videoPicture));
+  if (result)
+    drm->AddProperty(plane, "COLOR_ENCODING", value);
+
+  std::tie(result, value) =
+      plane->GetPropertyValue("COLOR_RANGE", DRMPRIME::GetColorRange(*videoPicture));
+  if (result)
+    drm->AddProperty(plane, "COLOR_RANGE", value);
```

Comment:
`COLOR_ENCODING` and `COLOR_RANGE` should be set regardless if `HDR_OUTPUT_METADATA` is supported or not, and the color encoding/range from video picture may also not be applicable for the gui plane.

Reply:
They are still set independently from the HDR metadata. I'm also not sure how it would be different for video/gui plane?

Reply:
This code will not always run because of `if (!settings->GetBool(SETTING_WINSYSTEM_IS_HDR_DISPLAY))` few lines up.

The color plane properties is used to inform the drm driver what color properties a non RGB plane has.
The gui plane color properties should match the EGL context, I think kodi always use an RGB plane as gui plane if I am not mistaken (I am unsure how HDR is handled in EGL).